routes/home.js

```diff
@@ -1,7 +1,36 @@
-module.exports = (req, res) => {
-  res.render('home', {
-    title: 'events',
-    calendarButton: true,
-    filterButtons: true
-  });
+const axios = require('axios');
+
+module.exports = async (req, res) => {
+  let url = `${process.env.URI}/events`;
+  try {
+    if (req.query.date_from) {
+      url =
+        url + `?date_from=${req.query.date_from}&date_to=${req.query.date_to}`;
+    } else {
+      const currentDate = new Date().toISOString().slice(0, 10);
+      url = url + `?date_from=${currentDate}`;
+    }
+    const eventsResponse = await axios.get(url);
+
+    let events = eventsResponse.data.filter(event => event['en']);
+
+    if (req.params.lang) {
+      events = eventsResponse.data.filter(event => event[req.params.lang]);
+    }
+
+    res.render('home', {
+      title: 'events',
+      events,
+      lang: req.params.lang || 'en',
+      calendarButton: true,
+      filterButtons: true,
+      english: req.params.lang !== 'ar',
+      arabic: req.params.lang === 'ar'
+    });
+  } catch (err) {
+    /* istanbul ignore next */
+    res.render('home', {
+      title: 'Not Found'
```

Comment:
When there's an error connecting to the OTP or this catch is hit for some other reason then there are a bunch of front-end console errors. This is because the 'home' render in the catch still links to the same JS files even though the DOM  elements that those files rely on is no longer there.

Reply:
Either run checks in the front end to see if the elements exist before setting event listeners, etc. or have a different template with no linked JS files for the error view.

Probably covered in the upcoming error page PR but still worth raising.

Reply:
I'm also gonna have a different PR where i handle every error in the app with the error view